industry_benchmarks/utils/tests/test_data_gathering.py

```diff
@@ -156,4 +200,190 @@ def test_get_sasa_diff(cmet_ligand_network):
 def test_charge_diff_no_diff(cmet_ligand_network):
     expected_diff = [0, 0, 0, 0]
     charge_diff = sorted([get_alchemical_charge_difference(edge) for edge in cmet_ligand_network.edges])
-    assert np.allclose(charge_diff, expected_diff)
\ No newline at end of file
+    assert np.allclose(charge_diff, expected_diff)
+
+
+def test_parse_alchemical_network(bace_network):
+    network = parse_alchemical_network(bace_network)
+    assert len(network.nodes) == 18
+    assert len(network.edges) == 22
+    assert isinstance(network, AlchemicalNetwork)
+
+
+def test_extract_ligand_network(bace_network):
+    network = parse_alchemical_network(bace_network)
+    ligand_network = extract_ligand_network(network)
+    assert isinstance(ligand_network, LigandNetwork)
+    # make sure they have the correct nodes and edges
+    assert len(network.nodes) == len(ligand_network.nodes) * 2
+    assert len(network.edges) == len(ligand_network.edges) * 2
+
+
+def test_load_results_file_wrong_file(bace_network, capsys):
+    """Make sure None is returned if we try and load the wrong file."""
+    result = load_results_file(bace_network)
+    assert result is None
+    assert "is a AlchemicalNetwork json, skipping" in capsys.readouterr().out
+
+def test_load_results_not_clean(bace_complete_results):
+    """Make sure an error is raised if the results have not been cleaned up."""
+    with pytest.raises(ValueError, match="has not been cleaned up please make sure you run the `results_cleanup.py` script first"):
+        _ = load_results_file(str(bace_complete_results / "results_0" / "solvent_spiro2_spiro1.json"))
+
+
+def test_get_estimate(cmet_results):
+    result = load_results_file(str(cmet_results / "results_0" / "lig_CHEMBL3402745_200_5_solvent_lig_CHEMBL3402744_300_4_solvent_solvent.json"))
+    ddg, error = get_estimate(result)
+    assert pytest.approx(ddg.m) == -13.506009
+    assert pytest.approx(error.m) == 0.0
+
+
+def test_get_transformation_name(bace_network, bace_cleaned_result):
+    cleaned_result = load_results_file(bace_cleaned_result)
+    network = parse_alchemical_network(bace_network)
+    transform_name = get_transform_name(cleaned_result, network)
+    assert transform_name == ("complex", "spiro10", "spiro6")
+
+
+@pytest.mark.parametrize("results_data, expected_output", [
+    pytest.param({
+        ("solvent", "lig_CHEMBL3402745_200_5", "lig_CHEMBL3402749_500_9"): [1, 2, 3],
+        ("solvent", "lig_CHEMBL3402754_40_14", "lig_CHEMBL3402761_1_21"): [1, 2, 3]
+    }, False, id="Not connected"),
+    pytest.param({
```

Comment:
Maybe add another test where we have both legs but not all three repeats in one of the legs?